clients/pkg/promtail/client/client.go

```diff
@@ -267,15 +436,28 @@ func (c *client) run() {
 	for {
 		select {
 		case e, ok := <-c.entries:
+
 			if !ok {
 				return
 			}
 			e, tenantID := c.processEntry(e)
+
+			wal, err := c.wal.getWAL(tenantID)
+			if err != nil {
+				level.Error(c.logger).Log("msg", "failed to get WAL", "err", err)
+				return
+			}
+			entryWriter.WriteEntry(e, wal, c.logger)
+
+			// after writing to wal, keep track of batch size to know when to cut the batch. That is either
+			// send it, or cut a segment in the wal
 			batch, ok := batches[tenantID]
 
 			// If the batch doesn't exist yet, we create a new one with the entry
 			if !ok {
-				batches[tenantID] = newBatch(c.maxStreams, e)
+				b := c.newBatch()
+				batches[tenantID] = b
+				_ = addEntryToBatch(b, e)
```

Comment:
Here and https://github.com/grafana/loki/blob/daa332547b94a0f323592b2b6fcfabae30e7e1ed/clients/pkg/promtail/client/client.go#L477-L477 we ignore the `addEntryToBatch` error. But further https://github.com/grafana/loki/blob/daa332547b94a0f323592b2b6fcfabae30e7e1ed/clients/pkg/promtail/client/client.go#L483-L483 checks and logs the error. Should we be checking and logging at all of these?

If we should be checking the error it might be useful to refactor so there's only a single call to `addEntryToBatch` at the end with the if statements ensuring the batch is sent/created ahead of it.

Reply:
Great idea! Refactoring the code to have a single call to `addEntryToBatch`